x/auth/tx/amino_aux_test.go

```diff
@@ -0,0 +1,155 @@
+package tx
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+func TestAminoAuxHandler(t *testing.T) {
+	privKey, pubkey, addr := testdata.KeyTestPubAddr()
+	interfaceRegistry := codectypes.NewInterfaceRegistry()
+	interfaceRegistry.RegisterImplementations((*sdk.Msg)(nil), &testdata.TestMsg{})
+	marshaler := codec.NewProtoCodec(interfaceRegistry)
+
+	txConfig := NewTxConfig(marshaler, []signingtypes.SignMode{signingtypes.SignMode_SIGN_MODE_AMINO_AUX})
+	txBuilder := txConfig.NewTxBuilder()
+
+	accountNumber := uint64(1)
+	chainId := "test-chain"
+	memo := "sometestmemo"
+	msgs := []sdk.Msg{testdata.NewTestMsg(addr)}
+	accSeq := uint64(2) // Arbitrary account sequence
+	timeout := uint64(10)
+
+	// any, err := codectypes.NewAnyWithValue(pubkey)
+	// require.NoError(t, err)
+
+	sigData := &signingtypes.SingleSignatureData{
+		SignMode: signingtypes.SignMode_SIGN_MODE_AMINO_AUX,
+	}
+
+	sig := signingtypes.SignatureV2{
+		PubKey:   pubkey,
+		Data:     sigData,
+		Sequence: accSeq,
+	}
```

Comment:
I would also move this to below `txBuilder.Set*`, just above `txBuilder.SetSignatures`